From: Robert Schwebel <r.schwebel@pengutronix.de>
Subject: util-linux-ng: fix libuuid handling in makefiles

The util-linux-ng tool mkswap can be built with uuid support. At the
moment the configure script automatically detects if libuuid is there or
not and stops the build if not. This is wrong, as libuuid is only
optionally needed for mkswap.

This patch does the following things:

* add an option --enable-uuid/--disable-uuid.
* if --enable-uuid is given, check if it is really there and break if not
* if --disable-uuid is given, don't build with uuid support
* if none of the above is given, auto-detecti if libuuid is there or not

Additionally, if libuuid is to be used, -luuid is added to mkswap,
swapon and swapoff. Although only mkswap needs functions from this lib,
the other tools link in libblkid which itself needs libuuid, so if not
linked in we get unresolved symbols.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

---
 configure.ac      |   39 ++++++++++++++++++++++++++++++++++++---
 mount/Makefile.am |    6 +++---
 2 files changed, 39 insertions(+), 6 deletions(-)

Index: util-linux-ng-2.14.1-rc2/configure.ac
===================================================================
--- util-linux-ng-2.14.1-rc2.orig/configure.ac
+++ util-linux-ng-2.14.1-rc2/configure.ac
@@ -136,10 +136,43 @@ AC_DEFUN([UTIL_CHECK_LIB], [
   AM_CONDITIONAL(AS_TR_CPP([HAVE_]suffix), [test [$have_]suffix = yes])
 ])
 
-UTIL_CHECK_LIB(uuid, uuid_is_null)
-if test "x$have_uuid" = xno; then
-  AC_MSG_WARN([uuid library is not found; mkswap(8) will not generate UUIDs])
+
+AC_MSG_CHECKING([whether to enable uuid support for mkswap(8)])
+AC_ARG_ENABLE(uuid,
+    AS_HELP_STRING([--enable-uuid], [enable uuid support for mkswap(8) @<:@default=auto@:>@]),
+	[case "$enableval" in
+	y | yes) CONFIG_UUID=yes;;
+	n | no)  CONFIG_UUID=no;;
+        auto)    CONFIG_UUID=auto;;
+	*)
+		AC_MSG_ERROR([--enable-uuid="$enableval" is not valid])
+		;;
+        esac],
+    [CONFIG_UUID=auto]
+)
+AC_MSG_RESULT([$CONFIG_UUID])
+
+if test $CONFIG_UUID = no; then
+	HAVE_UUID=false
+	UUID_LIBS=
+else
+	UTIL_CHECK_LIB(uuid, uuid_is_null)
+	if test $CONFIG_UUID = yes && test x$have_uuid = xno; then
+		AC_MSG_ERROR([uuid library is not found but --enable-uuid given; please install libuuid])
+	fi
+	if test $CONFIG_UUID = auto && test x$have_uuid = xno; then
+  		AC_MSG_WARN([uuid library is not found; mkswap(8) will not generate UUIDs])
+		HAVE_UUID=no
+		UUID_LIBS=
+	fi
+	if test x$have_uuid = xyes; then
+		HAVE_UUID=true
+		UUID_LIBS=-luuid
+	fi
 fi
+AM_CONDITIONAL(HAVE_UUID, $HAVE_UUID)
+AC_SUBST(UUID_LIBS)
+
 
 UTIL_CHECK_LIB(util, openpty)
 UTIL_CHECK_LIB(termcap, tgetnum)
Index: util-linux-ng-2.14.1-rc2/mount/Makefile.am
===================================================================
--- util-linux-ng-2.14.1-rc2.orig/mount/Makefile.am
+++ util-linux-ng-2.14.1-rc2/mount/Makefile.am
@@ -36,9 +36,9 @@ losetup_SOURCES = lomount.c sundries.c x
 losetup_CPPFLAGS = -DMAIN $(AM_CPPFLAGS)
 
 
-mount_LDADD = $(LDADD_common)
-umount_LDADD = $(LDADD_common)
-swapon_LDADD = $(LDADD_common)
+mount_LDADD = $(LDADD_common) $(UUID_LIBS)
+umount_LDADD = $(LDADD_common) $(UUID_LIBS)
+swapon_LDADD = $(LDADD_common) $(UUID_LIBS)
 
 LDADD_common =
 LDADD_common_static =
